lib/compile.js

```diff
@@ -23,10 +23,12 @@ module.exports = {
         if (stats.compilation.errors.length) {
           throw new Error('Webpack compilation error, see above');
         }
-        const outputPath = stats.compilation.compiler.outputPath;
-        this.webpackOutputPath = outputPath;
-        this.originalServicePath = this.serverless.config.servicePath;
-        this.serverless.config.servicePath = outputPath;
+        if (!this.options.location) {
+          const outputPath = stats.compilation.compiler.outputPath;
+          this.webpackOutputPath = outputPath;
+          this.originalServicePath = this.serverless.config.servicePath;
+          this.serverless.config.servicePath = outputPath;
+        }
```

Comment:
* With adding the compile call the serverless-offline location option is no longer necessary
* For backwards compatibility we can skip these lines when the location option is present